pkg/slack/app.go

```diff
@@ -41,52 +48,93 @@ func (a *App) Disabled() bool {
 	return a.disabled
 }
 
-func (a *App) GetChannels(ctx context.Context, repo string) ([]string, error) {
+// Format of channel info string: "<channel_Id>:<conclusion_1>,<conclusion_2>"
+func toChannelInfoString(channelInfo ChannelInfo) string {
+	if len(channelInfo.conclusions) == 0 {
+		return channelInfo.channelID
+	}
+	conclusionsString := strings.Join(channelInfo.conclusions, ",")
+	return channelInfo.channelID + ":" + conclusionsString
+}
+
+func (a *App) GetChannels(ctx context.Context, repo string) ([]ChannelInfo, error) {
 	data, err := a.store.Get(ctx, kvNamespace, repo)
 	if err != nil {
 		return nil, err
 	} else if data == "" {
 		return nil, nil
 	}
-	return strings.Split(data, ";"), nil
+	channelInfoStrings := strings.Split(data, ";")
+	channelInfos := []ChannelInfo{}
+
+	for _, channelString := range channelInfoStrings {
+		channelID, conclusionsString, _ := strings.Cut(channelString, ":")
+		consclusions := []string{}
+		for _, conclusion := range strings.Split(conclusionsString, ",") {
+			if len(conclusion) > 0 {
+				consclusions = append(consclusions, conclusion)
+			}
+		}
+		channelInfos = append(channelInfos, ChannelInfo{
+			channelID:   channelID,
+			conclusions: consclusions,
+		})
+	}
+
+	return channelInfos, nil
 }
 
-func (a *App) AddChannel(ctx context.Context, repo string, channelID string) error {
-	channelIDs, err := a.GetChannels(ctx, repo)
+func (a *App) AddChannel(ctx context.Context, repo string, channelInfo ChannelInfo) error {
+	channelInfos, err := a.GetChannels(ctx, repo)
 	if err != nil {
 		return err
 	}
 
-	for _, c := range channelIDs {
-		if c == channelID {
+	// Ref: https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#create-a-check-run--parameters
+	supportedConclusions := []string{"action_required", "cancelled", "failure", "neutral", "success", "skipped", "stale", "timed_out"}
+	unsupportedConclusions := []string{}
+	for _, c := range channelInfo.conclusions {
+		if !slices.Contains(supportedConclusions, c) {
+			unsupportedConclusions = append(unsupportedConclusions, c)
+		}
+	}
+
+	if len(unsupportedConclusions) > 0 {
+		return fmt.Errorf("unsupported conclusions: %s", strings.Join(unsupportedConclusions, ", "))
+	}
+
+	newChannelInfoStrings := []string{}
+	for _, c := range channelInfos {
+		if c.channelID == channelInfo.channelID {
```

Comment:
Should probably change duplication check into replacing it with with new conclusions, to support updating existing subscription directly?

Reply:
> Should probably change duplication check into replacing it with with new conclusions, to support updating existing subscription directly?

@kiootic Yes, I think this makes sense

cc @limouren